src/scib_metrics/utils/_pcr.py

```diff
@@ -38,6 +40,10 @@ def principal_component_regression(
         raise ValueError("Dimension mismatch: X must be 2-dimensional.")
     if X.shape[0] != covariate.shape[0]:
         raise ValueError("Dimension mismatch: X and batch must have the same number of samples.")
+    if categorical:
+        covariate = np.asarray(pd.Categorical(covariate).codes)
+    else:
+        covariate = np.asarray(covariate)
```

Comment:
```suggestion
        covariate = np.asarray(covariate)
        covariate = covariate.reshape((covariate.shape[0], 1))
```

Reply:
same here! I think we should take care of all input validations in a future PR